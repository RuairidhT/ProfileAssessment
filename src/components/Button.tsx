type Props = {
  title: string;
};

export const Button = ({ title }: Props) => {
  return (
    <button className="bg-indigo-600 dark:bg-slate-800 hover:dark:bg-slate-500 hover:bg-indigo-800 text-white py-2 px-4 rounded-full w-full">
      {title}
    </button>
  );
};
