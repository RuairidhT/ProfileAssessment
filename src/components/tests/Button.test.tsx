import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { Button } from '../Button';

describe('Button', () => {
  it('renders the button with the given title', () => {
    const title = 'Click me';
    const { getByText } = render(<Button title={title} />);

    const buttonElement = getByText(title);
    expect(buttonElement).toBeInTheDocument();
  });

  it('has the correct initial class names', () => {
    const title = 'Click me';
    const { getByText } = render(<Button title={title} />);

    const buttonElement = getByText(title);
    expect(buttonElement).toHaveClass('bg-indigo-600');
    expect(buttonElement).toHaveClass('dark:bg-slate-800');
    expect(buttonElement).toHaveClass('hover:dark:bg-slate-500');
    expect(buttonElement).toHaveClass('hover:bg-indigo-800');
    expect(buttonElement).toHaveClass('text-white');
    expect(buttonElement).toHaveClass('py-2');
    expect(buttonElement).toHaveClass('px-4');
    expect(buttonElement).toHaveClass('rounded-full');
    expect(buttonElement).toHaveClass('w-full');
  });

  it('should match snapshot', () => {
    const title = 'Click me';
    const { container } = render(<Button title={title} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
