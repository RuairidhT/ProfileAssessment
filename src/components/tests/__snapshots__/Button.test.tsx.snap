// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should match snapshot 1`] = `
<button
  class="bg-indigo-600 dark:bg-slate-800 hover:dark:bg-slate-500 hover:bg-indigo-800 text-white py-2 px-4 rounded-full w-full"
>
  Click me
</button>
`;
